lib/max_subarray.rb

```diff
@@ -1,8 +1,166 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def max_sub_array(nums)
```

Comment:
The brute force solution here works, but is not dynamic and is a bit inefficient.  It looks like you were on the way to a dynamic solution, but not quite there.